scripts/compute_sample_luminosity.py

```diff
@@ -0,0 +1,109 @@
+#! /usr/bin/env python
+""" Simple script to compute the luminosity of a set of samples """
+
+# Storms package
+import sys
+sys.path.append('/nfs/soft/python/python-2.7.5-sl6_amd64_gcc44/lib/python2.7/site-packages/storm-0.20-py2.7-linux-x86_64.egg')
+sys.path.append('/nfs/soft/python/python-2.7.5-sl6_amd64_gcc44/lib/python2.7/site-packages/MySQL_python-1.2.3-py2.7-linux-x86_64.egg')
+
+import subprocess
+import argparse
+from SAMADhi import Sample, DbStore
+
+def get_options():
+    parser = argparse.ArgumentParser(description='Compute luminosity of a set of samples.')
+
+    parser.add_argument('-i', '--id', type=int, nargs='+', dest='ids', help='IDs of the samples', metavar='ID')
+    parser.add_argument('--name', type=str, nargs='+', dest='names', help='Names of the samples', metavar='NAME')
+
+    parser.add_argument('--local', dest='local', action='store_true', help='Run brilcalc locally instead of on lxplus')
+
+    options = parser.parse_args()
+
+    if options.ids is None and options.names is None:
+        parser.error('You must specify at least one sample id or sample name.')
+
+    if options.ids is None:
+        options.ids = []
+
+    if options.names is None:
+        options.names = []
+
+    return options
+
+def get_sample(id, name):
+
+    dbstore = DbStore()
+
+    if id is not None:
+        result = dbstore.find(Sample, Sample.sample_id == id)
+    elif name is not None:
+        result = dbstore.find(Sample, Sample.name.like(unicode(name.replace('*', '%').replace('?', '_'))))
+
+    return result.one()
+
+def parse_luminosity_csv(result):
+    """ Parse the CSV file produced by brilcalc, and return the total recorded luminosity in /pb """
+    import csv
+    import StringIO
+
+    f = StringIO.StringIO(result)
+
+    lumi = 0
+    reader = csv.reader(f, delimiter=',')
+    for row in reader:
+        if row[0][0] == '#':
+            continue
+        lumi += float(row[-1])
+
+    return lumi / 1000. / 1000.
+
+def compute_luminosity(sample, options):
+    print("Computing luminosity for %r") % str(sample.name)
+
+    lumi = 0
+    if not options.local:
+        print("Running brilcalc on lxplus... You'll probably need to enter your lxplus password in a moment")
+        print('')
+
+        cmds = ['brilcalc', 'lumi', '--normtag', '/afs/cern.ch/user/c/cmsbril/public/normtag_json/OfflineNormtagV1.json', '--output-style', 'csv', '-i', '"%s"' % str(sample.processed_lumi.replace('"', ''))]
+        cmd = 'export PATH="/afs/cern.ch/user/s/sbrochet/.local/bin:/afs/cern.ch/cms/lumi/brilconda-1.0.3/bin:$PATH"; export PYTHONPATH="/afs/cern.ch/user/s/sbrochet/.local/lib/python2.7"; ' + ' '.join(cmds)
+        ssh_cmds = ['ssh', 'lxplus.cern.ch', cmd]
```

Comment:
The local and lxplus username must be the same... it would be nice to add an option to specify the remote user. Maybe to be combined with the previous comment.